StevesSixLeggedSprinters/BangazonAPI
# Kk training

## Status
Choose:
**HOLD**

## Migrations
YES 

## Description
I created the training program model (TrainingProgram.cs) in order to create a table in the database for all training programs. I also created a training program controller (TrainingProgramController.cs) so we are able to GET a list of all training programs, GET a single training program, POST (create) a new training program, PUT (edit) a specific training program, and DELETE a specific training program (but only IF the start date of the program is a future date. 
Also modified BangazonAPIContext file to set up a relationship between the database and the computer table.



## Todos


## Deploy Notes
Notes regarding deployment the contained body of work.  These should note any
db migrations, etc.

## Steps to Test or Reproduce
```I have set up the following methods: GET, POST, PUT, DELETE
Follow these step by step in order to test.

git fetch --a  (fetches all branches from GitHub)
git branch -avv (display all branches)
git checkout kk-training (switch into my branch)
git pull origin kk-training (pull my code down to test)
dotnet ef migrations add Initial (migrations need to be built)
dotnet ef database update (update the database since a new migration has been built)
open bangazon.db (open database)
dotnet run

Make sure to have Postman installed: https://www.getpostman.com/
Use the URL http://localhost:5000/api/trainingprogram/ 

In Postman, 'POST' (add) a couple training programs to the database to start testing.
Create some training programs with a starting date earlier than today's current date.
Also, create programs with starting dates in the future.

**POST** Example: (with a future start date - this should be able to be deleted from the db)
{
    "TrainingProgramName": "Bangazon Orientation",
    "StartDate": "05-05-2018",
    "EndDate": "05-05-2020",
    "MaxAttendees": 15
}

**POST** example: (with a starting date that occurred in the past - this should NOT be able to be deleted from the db)
{
    "TrainingProgramName": "Bangazon Orientation",
    "StartDate": "05-05-2015",
    "EndDate": "05-05-2015",
    "MaxAttendees": 20
}

To view all training programs:
Select **GET**
http://localhost:5000/api/trainingprograms

To view a single computer:
Select **GET**
http://localhost:5000/api/trainingprograms/{id}

To edit a single computer:
Select **PUT**
http://localhost:5000/api/trainingprograms/{id}

To delete a single training program:
Select **DELETE** (remember to test programs with both a future starting date, and a program with a starting date that is prior to today's date.
http://localhost:5000/api/trainingprograms/{id}```


## Impacted Areas in Application

* BangazonAPIContext.cs
* TrainingProgram.cs
*TrainingProgramController.cs

#9
